pkgs/development/misc/resholve/deps.nix

```diff
@@ -0,0 +1,150 @@
+{ stdenv
+, fetchFromGitHub
+, makeWrapper
+, # re2c deps
+  autoreconfHook
+, # py-yajl deps
+  git
+, # oil deps
+  readline
+, cmark
+, python27
+, file
+, glibcLocales
+, oilPatches ? [ ]
+}:
+
+/*
+Notes on specific dependencies:
+- if/when python2.7 is removed from nixpkgs, this may need to figure
+  out how to build oil's vendored python2
+- I'm not sure if glibcLocales is worth the addition here. It's to fix
+  a libc test oil runs. My oil fork just disabled the libc tests, but
+  I haven't quite decided if that's the right long-term call, so I
+  didn't add a patch for it here yet.
+*/
+
+rec {
+  # had to add this as well; 1.3 causes a break here; sticking
+  # to oil's official 1.0.3 dep for now.
+  re2c = stdenv.mkDerivation rec {
+    pname = "re2c";
+    version = "1.0.3";
+    sourceRoot = "${src.name}/re2c";
+    src = fetchFromGitHub {
+      owner = "skvadrik";
+      repo = "re2c";
+      rev = version;
+      sha256 = "0grx7nl9fwcn880v5ssjljhcb9c5p2a6xpwil7zxpmv0rwnr3yqi";
+    };
+    nativeBuildInputs = [ autoreconfHook ];
+    preCheck = ''
+      patchShebangs run_tests.sh
+    '';
+  };
+
+  py-yajl = python27.pkgs.buildPythonPackage rec {
+    pname = "oil-pyyajl";
+    version = "unreleased";
+    src = fetchFromGitHub {
+      owner = "oilshell";
+      repo = "py-yajl";
+      rev = "eb561e9aea6e88095d66abcc3990f2ee1f5339df";
+      sha256 = "17hcgb7r7cy8r1pwbdh8di0nvykdswlqj73c85k6z8m0filj3hbh";
+      fetchSubmodules = true;
+    };
+    # just for submodule IIRC
+    nativeBuildInputs = [ git ];
+  };
+
+  # resholve's primary dependency is this developer build of the oil shell.
+  oildev = python27.pkgs.buildPythonPackage rec {
```

Comment:
You should use declare `buildPythonPackage` as a parameter instead, even if this package is only build using Python 2.7. To make this build fails with Python 3, use something like this: https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/python-modules/venusian/default.nix#L12 (but with `disabled = !isPy27` instead).

Reply:
BTW, all those dependencies should be separate packages inside `pkgs/development/python-modules`, and the main package should use them as `buildDependencies`.

Reply:
BTW, having oilshell and maybe oildev available in `all-packages` is probably a good idea, since those packages are kinda interesting.

Reply:
You could, for example, open a PR with only `oildshell`, `oildev` and maybe `py-yajl`, making it easier to review only those dependencies. Afterwards you could make this PR depends on those dependencies.

Reply:
> BTW, all those dependencies should be separate packages inside `pkgs/development/python-modules`, and the main package should use them as `buildDependencies`. BTW, having oilshell and maybe oildev available in `all-packages` is probably a good idea, since those packages are kinda interesting. You could, for example, open a PR with only `oildshell`, `oildev` and maybe `py-yajl`, making it easier to review only those dependencies. Afterwards you could make this PR depends on those dependencies.

I have reservations about this. 

I'm happy to break these out if there's a clear imperative, but they are all pretty idiomatic (and all ultimately subservient to the oildev package itself). I don't anticipate them adding standalone value to nixpkgs in the short run.

1. The Oil shell already has a binary user package included here: https://github.com/NixOS/nixpkgs/blob/master/pkgs/shells/oil/default.nix
2. oildev isn't (for now) a _real thing_ that the Oil project intentionally publishes--it's a fiction created by forcibly patching-and-packaging Oil's python code in order to build resholve
    - Because these Python APIs aren't intentionally published for external consumption and Oil is still a fast-changing young project, they aren't stable. I'd be very surprised if more than one consumer could share a single version of it for long.
    - AFAIK resholve is the only consumer of this. No one's asked me about using it.
    - resholve still needs [~140 lines of code](https://github.com/abathur/resholve/blob/d990fc06b504d761ff811bd1ad8075bdceb61dd0/resholve#L23-L161) just to monkeypatch all of oil's packages into a unified oil.* namespace. If anyone used it without this shim, all of its subpackages would leak into the root Python package namespace.
    - I can't guarantee any Oil code paths that resholve doesn't exercise actually work.
    - Because resholve is very tightly coupled to Oil's internal Python APIs, there's a fair chance any update of this would break it.
3. [py-yajl](https://github.com/oilshell/py-yajl) is Oil's fork of https://github.com/rtyler/py-yajl. They've modified it, dropped tests they don't need, and cut code they don't use. I have no idea how suitable it is or isn't for general use. Oil has made no attempt to promote their py-yajl fork, it has no stars, and I have zero reason to expect that Oil would support it in any way.

Reply:
> The Oil shell already has a binary user package included here: https://github.com/NixOS/nixpkgs/blob/master/pkgs/shells/oil/default.nix

I did a search for `oilshell` and didn't found anything, TIL.